src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -24,13 +25,28 @@ public CompanyProfileService(Logger logger, CompanyProfileRepository companyProf
      * @return a company profile if one with such a company number exists, otherwise an empty
      *     optional
      */
-    public Optional<CompanyProfileDao> get(String companyNumber) {
+    public Optional<CompanyProfileDocument> get(String companyNumber) {
         logger.trace(String.format("DSND-374: GET company profile with number %s", companyNumber));
-        Optional<CompanyProfileDao> companyProfileDao = companyProfileRepository
+        Optional<CompanyProfileDocument> companyProfileDao = companyProfileRepository
                 .findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(companyNumber);
 
         logger.trace(String.format("DSND-374: Company profile with number %s retrieved: %s",
                 companyNumber, companyProfileDao));
         return companyProfileDao;
     }
+
+    /**
+     * Updated insolvency links in company profile.
+     * @param companyProfileRequest company Profile information {@link CompanyProfile}
+     */
+    public void updateInsolvencyLink(final CompanyProfile companyProfileRequest) {
+        CompanyProfileDocument companyProfile = companyProfileRepository
+                .findCompanyProfileDaoByCompanyProfile_Data_CompanyNumber(
+                        companyProfileRequest.getData().getCompanyNumber()).get();
```

Comment:
This assumes a company profile matching that company number will already exist in the DB, not sure if in scope for your ticket but the Optional should be checked and an error thrown if `isEmpty()`;

Reply:
Good call